infra/blueprint-test/modules/bpt/bpt.go

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Package bpt defines a blueprint and implements default stages and execution order for a blueprint test
+
+package bpt
+
+import (
+	"github.com/mitchellh/go-testing-interface"
+
+	"github.com/GoogleCloudPlatform/cloud-foundation-toolkit/infra/blueprint-test/modules/utils"
+	"github.com/stretchr/testify/assert"
+)
+
+// Blueprint represents a config that can be setup, applied, verified and torndown.
+type Blueprint interface {
+	Setup()
+	Apply()
+	Verify(*assert.Assertions)
+	Teardown()
+}
+
+// BlueprintTest implements a generic blueprint
+type BlueprintTest struct {
+	Setup    func()
+	Apply    func()
+	Verify   func(*assert.Assertions)
+	Teardown func()
+}
+
+func (b *BlueprintTest) DefineSetup(setup func()) {
```

Comment:
Since we also have the concept of "setup" steps (that happen before tests) this stage name is overloaded. Perhaps DefineInit?

Reply:
switched to init